otelcol/command.go

```diff
@@ -16,14 +16,29 @@ import (
 // Any URIs specified in CollectorSettings.ConfigProviderSettings.ResolverSettings.URIs
 // are considered defaults and will be overwritten by config flags passed as
 // command-line arguments to the executable.
+//
+// Deprecated: [v0.103.0] use NewCommandMustSetProvider instead
 func NewCommand(set CollectorSettings) *cobra.Command {
+	return commandHelper(set, false)
+}
+
+// NewCommandMustSetProvider constructs a new cobra.Command using the given CollectorSettings.
+// Any URIs specified in CollectorSettings.ConfigProviderSettings.ResolverSettings.URIs
+// are considered defaults and will be overwritten by config flags passed as
+// command-line arguments to the executable.
+// At least one provider or converter must be supplied via CollectorSettings.ConfigProviderSettings.ResolverSettings.
```

Comment:
```suggestion
// At least one Provider must be supplied via CollectorSettings.ConfigProviderSettings.ResolverSettings.ProviderFactories.
```
Supplying 0 Providers isn't valid, so I think we should call that out here.